compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.test.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "VISQErrorApproximator.h"
+
+#include <json.h>
+#include <iostream>
+#include <fstream>
+#include <unistd.h>
+#include <gtest/gtest.h>
+
+namespace
+{
+
+void writeDataToFile(const std::string &path, const std::string &data)
+{
+  std::ofstream file;
+  file.open(path);
+  file << data;
+  file.close();
+}
+
+void makeTemporaryFile(char *name_template)
+{
+  int fd = mkstemp(name_template);
+  if (fd == -1)
+  {
+    throw std::runtime_error{"mkstemp failed"};
+  }
+}
+
+} // namespace
+
+TEST(CircleMPQSolverVISQErrorApproximatorTest, verifyResultsTest)
+{
+  static std::string errors_key = "error";
+  static std::string layer_key = "layer_0";
+  static float layer_error = 0.5f;
+  // trivial json with a single layer
+  Json::Value error_data;
+  Json::Value layer_data;
+  layer_data[layer_key] = layer_error;
+  error_data[errors_key].append(layer_data);
+
+  Json::StreamWriterBuilder builder;
+  auto data = Json::writeString(builder, error_data);
+  char path[] = "/tmp/VISQErrorApproximator-TEST-XXXXXX";
+  makeTemporaryFile(path);
```

Comment:
use system API to get temporary file

Reply:
@seanshpark 
Unfortunately i failed to query the name (which can be transferred to constructor of `VISQErrorAppriximator`) of temporary file generated in this way. `tmpnam` is not recommended to use though.

Reply:
@seanshpark 
Ahh. Sorry. Didn't saw a method to extract name of tempfile, i'll try. Thank you.